cfme/tests/ansible/test_embedded_ansible_services.py

```diff
@@ -707,3 +708,50 @@ def _revert():
     view = navigate_to(ansible_service, "Details")
     assert verbosity[0] == view.provisioning.details.get_text_of("Verbosity")
     assert verbosity[0] == view.retirement.details.get_text_of("Verbosity")
+
+
+@pytest.mark.tier(1)
+def test_service_ansible_playbook_vault_credentials(
+    appliance,
+    request,
+    ansible_catalog_item,
+    ansible_service_catalog,
+    ansible_service_request,
+    ansible_service,
+):
+    """
+    Add vault password and test in the playbook that encrypted yml can be
+    decrypted.
+
+    Polarion:
+        assignee: sbulage
+        casecomponent: Ansible
+        initialEstimate: 1/2h
+        tags: ansible_embed
+    """
+    creds = conf.credentials['vault_creds']['password']
+    creds_dict = {"vault_password": creds}
+    vault_creds = appliance.collections.ansible_credentials.create(
+        "Vault_Credentials_{}".format(fauxfactory.gen_alpha()), "Vault", **creds_dict
+    )
+
+    with update(ansible_catalog_item):
+        ansible_catalog_item.provisioning = {
+            "playbook": "dump_secret_variable_from_vault.yml",
+            "vault_credential": vault_creds.name,
+        }
+
+    @request.addfinalizer
+    def _revert():
+        with update(ansible_catalog_item):
```

Comment:
I probably miss some point. could you help me understand why you don't remove created vault credential but just unset created credential ?

Reply:
Adding credentials removal in finalizer.

Reply:
Updated with credentials finalizer.